crates/subspace-farmer/src/cluster/cache.rs

```diff
@@ -361,99 +361,34 @@ where
     caches_details
         .iter()
         .map(|cache_details| async move {
-            // Initialize with pending future so it never ends
-            let mut processing =
-                FuturesUnordered::<Pin<Box<dyn Future<Output = ()> + Send>>>::from_iter([
-                    Box::pin(pending()) as Pin<Box<_>>,
-                ]);
-            let subscription = nats_client
-                .queue_subscribe(
-                    ClusterCacheWritePieceRequest::SUBJECT
-                        .replace('*', &cache_details.cache_id_string),
-                    cache_details.cache_id_string.clone(),
+            let process = |ClusterCacheWritePieceRequest {
+                               offset,
+                               piece_index,
+                               piece,
+                           }| async move {
```

Comment:
This is a clear example where I wouldn't use destructuring in the argument position, it leads to awkward formatting, just do it inside of the function.

Reply:
fixed